### PR TITLE
[Driver][SYCL][FPGA] Process aoco as regular archives for emulation

### PR DESCRIPTION
**Pull request by mdtoguchi**
AOCO archives are treated in a special manner for FPGA consumption.  Much
of this functionality is handled within the 'aoc' tool.  Now that we have
moved away from using 'aoc' for emulation (using opencl-aot), modify the
behavior to treat aoco archives are regular archives only and process
through opencl-aot.

### CODE REVIEW

File: clang/test/Driver/sycl-intelfpga-aoco-emu.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// UNSUPPORTED: system-windows
+
+/// -fintelfpga static lib (aoco)
```
Comment (AGindinson) [inline clang/test/Driver/sycl-intelfpga-aoco-emu.cpp:3]: If this flow is similar to what we have for non-FPGA-HW targets when treating static archives, adding the first check block could be sufficient (perhaps within `sycl-intelfpga-aoco.cpp`, although I understand that the archive would need to be re-compiled). Provided that existing tests provide sufficient coverage for the "regular" flow, of course.

**Review by AGindinson — APPROVED** [review]

File: clang/test/Driver/sycl-intelfpga-aoco-emu.cpp (hunk 1 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-intelfpga-aoco-emu.cpp:3]: Yes, this flow should be the same as other static archives.  Using sycl-intelfpga-aoco.cpp should be fine without re-compiling the archive - it's just a matter of what command line option is used.

File: clang/test/Driver/sycl-intelfpga-aoco.cpp [hunk 2]
```diff
@@ -100,3 +100,52 @@
 // CHK-FPGA-AOCO-WIN: llc{{.*}} "-filetype=obj" "-o" "[[FINALOBJW:.+\.obj]]" "[[FINALBC]]"
 // CHK-FPGA-AOCO-LIN: ld{{.*}} "[[INPUTLIB]]" {{.*}} "[[FINALOBJL]]"
 // CHK-FPGA-AOCO-WIN: link.exe{{.*}} "{{.*}}[[INPUTLIB]]" {{.*}} "[[FINALOBJW]]"
+
+/// aoco archive check with emulation
+// RUN:  %clangxx -target x86_64-unknown-linux-gnu -fsycl -fno-sycl-device-lib=all -fintelfpga %t_aoco.a %s -ccc-print-phases 2>&1 \
+  // RUN:  | FileCheck -check-prefix=CHK-FPGA-AOCO-PHASES-EMU %s
```
Comment (AGindinson) [inline clang/test/Driver/sycl-intelfpga-aoco.cpp:106]:
```suggestion
// RUN:  %clangxx -target x86_64-unknown-linux-gnu -fsycl -fno-sycl-device-lib=all -fintelfpga %t_aoco.a %s -ccc-print-phases 2>&1 \
// RUN:  | FileCheck -check-prefix=CHK-FPGA-AOCO-PHASES-EMU %s
```

**Review by AGindinson — APPROVED** [review]
Thanks!

**Pull request merged**